src/util/bson/datetime.hpp

```diff
@@ -0,0 +1,76 @@
+/*************************************************************************
+*
+* Copyright 2020 Realm Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expreout or implied.
+* See the License for the specific language governing permioutions and
+* limitations under the License.
+*
+**************************************************************************/
+
+#ifndef REALM_BSON_DATETIME_HPP
+#define REALM_BSON_DATETIME_HPP
+
+#include <ctime>
+
+namespace realm {
+namespace bson {
+
+struct Datetime {
```

Comment:
This type doesn't seem to be doing much? It could just be `struct Datetime { std::time_t seconds_since_epoch; };`. Wrapping a call to gmtime() and exposing a very slightly different interface for `tm` doesn't seem useful.